kyo-reactive-streams/shared/src/main/scala/kyo/interop/reactive-streams/StreamSubscription.scala

```diff
@@ -0,0 +1,148 @@
+                                .andThen(Loop.done[Long, Chunk[V] | StreamFinishState](
+                                    values.drop(requesting.intValue)
+                                ))
+                    case Absent =>
+                        IO(Loop.done[Long, Chunk[V] | StreamFinishState](StreamFinishState.StreamComplete))
+                }
+            }
+
+        Loop[Chunk[V] | StreamFinishState, StreamFinishState, Async & Poll[Chunk[V]]](Chunk.empty[V]) {
+            case leftOver: Chunk[V] =>
+                for
+                    requestingResult <- (requestChannel.poll(): @unchecked) match
+                        case Result.Success(Present(requesting)) => IO(Result.Success(requesting))
+                        case Result.Success(Absent)              => requestChannel.takeFiber().safe.getResult
+                        case error: Result.Error[Closed]         => IO(error)
+                    outcome <- requestingResult match
+                        case Result.Success(requesting) =>
+                            if requesting < leftOver.size then
+                                IO(leftOver.take(requesting.intValue).foreach(subscriber.onNext(_)))
+                                    .andThen(Loop.continue[StreamFinishState | Chunk[V], StreamFinishState, Async & Poll[Chunk[V]]](
+                                        leftOver.drop(requesting.intValue)
+                                    ))
+                            else
+                                IO(leftOver.foreach(subscriber.onNext(_)))
+                                    .andThen(loopPoll(requesting - leftOver.size))
+                                    .map(Loop.continue[StreamFinishState | Chunk[V], StreamFinishState, Async & Poll[Chunk[V]]](_))
+                        case Result.Fail(_) =>
+                            IO(Loop.continue[StreamFinishState | Chunk[V], StreamFinishState, Async & Poll[Chunk[V]]](
+                                StreamFinishState.StreamCanceled
+                            ))
+                        case Result.Panic(exception) => IO(throw exception).andThen(Loop.continue[
+                                StreamFinishState | Chunk[V],
+                                StreamFinishState,
+                                Async & Poll[Chunk[V]]
+                            ](StreamFinishState.StreamCanceled))
+                yield outcome
+            case state: StreamFinishState => Loop.done(state)
+        }
+    end poll
+
+    private[reactivestreams] def consume(
+        using
+        emitTag: Tag[Emit[Chunk[V]]],
+        pollTag: Tag[Poll[Chunk[V]]],
+        frame: Frame,
+        safepoint: Safepoint
+    ): Fiber[Nothing, StreamFinishState] < (IO & Ctx) =
+        boundary { (trace, context) =>
+            val fiber = Fiber.fromTask(IOTask(Poll.run(stream.emit)(poll).map(_._2), safepoint.copyTrace(trace), context))
```

Comment:
`IOTask` is an internal detail of `Fiber`/`Async`. Ideally, we should rely on the existing APIs instead of using it directly

Reply:
this is just a copy of `Async._run` with boundary that does not care about `Abort[E]` (`E =:= Nothing` in this case). Maybe after `https://github.com/getkyo/kyo/pull/906` is merged I can use `_run` instead